scalafix-core/shared/src/main/scala/scalafix/internal/rule/ExplicitResultTypes.scala

```diff
@@ -112,11 +116,16 @@ case class ExplicitResultTypes(
       def isImplicit: Boolean =
         defn.hasMod(mod"implicit") && !isImplicitly(body)
 
-      def isLocal: Boolean =
+      def hasParentWihTemplate: Boolean =
         defn.parent.exists(_.is[Template])
 
-      isImplicit || {
-        isLocal &&
+      def isLocal = nm.symbol match {
+        case Some(value) => value.isInstanceOf[scala.meta.Symbol.Local]
+        case None => false
+      }
+
+      isImplicit && (if (printLocalTypes) !isLocal else true) || {
```

Comment:
I would prefer to keep this boolean condition as simple as possible and delegate logic to local defs. We can move this if/else into `isLocal`